packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -116,7 +116,7 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
     super.beginActivity(newActivity);
     _currentDrag?.dispose();
     _currentDrag = null;
-    if (!activity.isScrolling)
+    if (!activity.isScrolling && !(activity is HoldScrollActivity))
```

Comment:
It takes me sometime to understand why this work. and I think this is the right fix.

For other reviewers, the crash happens when holdaction is interrupted by other action during initialization. The logic in scrollable relying on atomicity of hold action. The current logic wrongly notify the listener which gives develop the chance to interrupted hold action

Reply:
Thank you! Yes. It's tricky to explain. Thank you for providing a clearer explanation. 
I've been thinking of is a as, it is notifying as idle based solely on whether or not the activity is scrolling. A hold activity is not scrolling, but it is not idle either.